vendor/plugins/edu.harvard.dce.paella.handoutDownloadPlugin/handout.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * #DCE MATT-1794, UI pluging for user to access one or more Mediapackage attachments
+ * of type "attachment/notes". For example, a PDF handout.
+ * #DCE-1840, Change handout download to a single button click
+ */
+Class ("paella.plugins.handoutDownloadPlugin", paella.ButtonPlugin,{
+	_attachementPath: ['matterhorn', 'episode', 'mediapackage', 'attachments', 'attachment'],
+	_attachments: [],
+	_downloadUrl:null,
+
+	getAlignment:function() { return 'right'; },
+	getSubclass:function() { return "handoutDownloadPlugin"; },
+	getIndex:function() { return 2030; },
+	getMinWindowSize:function() { return 350; },
+	getName:function() { return "edu.harvard.dce.paella.handoutDownloadPlugin"; },
+	getDefaultToolTip:function() { return base.dictionary.translate("Download Class Handout"); },
+
+     showUI: function(){
+        // Show if handout attachment exist
+        if (this._downloadUrl != null ) {
+           this.parent();
+        }
+    },
+
+    setup: function () {
+       // make sure there is something to put in the something else
+       if ((this._downloadUrl != null) && ($(this.container).length == 1)) {
+          $(this.container).append(this.createAnchorLink());
+       }
+    },
+
+    checkEnabled: function (onSuccess) {
+        // retrieve any attached handouts (type "attachment/notes")
+        // FYI, live events do not contain the attachments section
+        if (this.pathExists(paella, this._attachementPath)) {
+          var attachments = paella.matterhorn.episode.mediapackage.attachments.attachment;
+          if (!(attachments instanceof Array)) {
+            attachments =[attachments];
+          }
+          this.setDownloadUrl(attachments);
+        }
+        // verify download url was set
+        var isenabled = (this._downloadUrl != null );
+        onSuccess(isenabled);
+    },
+
+    action:function(button) {
+       if (this._downloadUrl != null) {
```

Comment:
I removed the extra check in action function, because, as you say, there better be a URL at that point, but left it at the setup method, just because it seems safer to make sure the timing is right. In theory, the check enabled should happen before the setup.